File: frontend/src/services/ai.ts
import api from './api';

export interface ChatResponse {
  reply: string;
  suggestions?: Array<{
    id: number;
    name: string;
    brand: string;
    slug: string;
    thumbnail_url?: string;
    price?: number;
    reason?: string;
  }>;
  follow_up_hint?: string;
}

export async function chat(message: string, sessionId: string, history?: string[]): Promise<ChatResponse> {
  const { data } = await api.post('/ai/chat', {
    message,
    session_id: sessionId,
    history,
  });
  return data as ChatResponse;
}

export interface RecommendResponse {
  suggestions: Array<{
    id: number;
    name: string;
    brand: string;
    slug: string;
    thumbnail_url?: string;
    price?: number;
    reason?: string;
  }>;
}

export async function recommend(query: string): Promise<RecommendResponse> {
  const { data } = await api.post('/ai/recommend', { query });
  return data as RecommendResponse;
}

File: backend/internal/dto/ai.go
package dto

// RecommendRequest is the input payload for the AI recommend endpoint.
type RecommendRequest struct {
	Message string   `json:"message"`
	History []string `json:"history,omitempty"`
	Limit   int      `json:"limit,omitempty"`
}

// RecommendSuggestion is a compact product card to show inside the chat.
type RecommendSuggestion struct {
	ID           uint    `json:"id"`
	Name         string  `json:"name"`
	Brand        string  `json:"brand"`
	Slug         string  `json:"slug"`
	ThumbnailURL string  `json:"thumbnail_url"`
	Price        float64 `json:"price"`
	Reason       string  `json:"reason"`
}

// RecommendResponse bundles suggestions and lightweight reasoning.
type RecommendResponse struct {
	Suggestions []RecommendSuggestion `json:"suggestions"`
	Reasoning   string                `json:"reasoning,omitempty"`
}
